startup.py

```diff
@@ -0,0 +1,340 @@
+
+import sgtk
+from sgtk import TankError
+from sgtk.platform import SoftwareLauncher, SoftwareVersion, LaunchInformation
+
+class MayaLauncher(SoftwareLauncher):
+    """
+    Handles launching Maya without having an active engine.
+    """
+    def scan_software(self, versions=None, display_name=None, icon=None):
+        """
+        Performs a scan for software installations.
+
+        :param versions: List of strings representing versions
+                         to search for. If set to None, search
+                         for all versions. A version string is
+                         DCC-specific but could be something
+                         like "2017", "6.3v7" or "1.2.3.52"
+        :param display_name: (optional) Specify label to use for all
+                             SoftwareVersions found.
+        :param icon: (optional) Specify icon to use for all
+                     SoftwareVersions found.
+
+        :returns: List of :class:`SoftwareVersion` instances
+        """
+        # First look for executables using the Autodesk Synergy registry.
+        sw_versions = _synergy_software_versions(
+            self.logger, versions, display_name, icon
+        )
+        if not sw_versions:
+            # Look for executables in paths formerly specified by the
+            # default configuration paths.yml file.
+            sw_versions = _default_path_software_versions(
+                self.logger, versions, display_name, icon
+            )
+        if not sw_versions:
+            self.logger.info(
+                "Unable to determine available SoftwareVersions for engine %s" %
+                self.engine_name
+            )
+            return []
+
+        return  sw_versions
+
+
+    def prepare_launch(self, exec_path, args, options, file_to_open=None):
+        """
+        Prepares the given software for launch
+
+        :param exec_path: Path to DCC executable to launch
+        :param args: Command line arguments as strings
+        :param options: DCC specific options to pass
+        :param file_to_open: (Optional) Full path name of a file to open on launch
+
+        :returns: LaunchInformation instance
+        """
+        required_env = {}
+        startup_path = os.path.join(self.disk_location, "startup")
+        sgtk.util.append_path_to_env_var("PYTHONPATH", startup_path)
+        required_env["PYTHONPATH"] = os.environ["PYTHONPATH"]
+        required_env["TANK_ENGINE"] = self.engine_name
+        required_env["TANK_CONTEXT"] = sgtk.context.serialize(self.context)
+        if file_to_open:
+            required_env["TANK_FILE_TO_OPEN"] = file_to_open
+
+        return LaunchInformation(exec_path, args, required_env)
+
+
+def _icon_from_executable(exec_path):
+    """
+    Find the application icon based on the executable path and
+    current platform.
+
+    :param exec_path: Full path to the executable
+
+    :returns: Full path to application icon as a string or None.
+    """
+    icon_base_path = ""
```

Comment:
seems like this method should have a bunch of debug info so in the case the icon cannot be found on someone's system, we can more easily diagnose why that is!